cicd/README.md

```diff
@@ -31,29 +7,45 @@ There are three phases of the CI/CD configuration. They are best explained in re
 
 ### 2 - CI/CD
 
-In order to trigger the application resources to be updated upon changes to the source code, we need CI/CD resources. This is accomplished by a CloudFormation template that defines a stack of resources, primarily including a CodeBuild project and a CodePipeline pipeline which update the [App Stack](#3---app). These CI/CD resources only need to be deployed once per deployable branch, `main` in our case (we might choose to create adhoc environments by launching a new CI/CD stack targeting that branch).
+In order to trigger the application resources to be updated upon changes to the source code, we need CI/CD resources. This is accomplished by a CloudFormation template that defines a stack of resources, primarily including a CodeBuild project and a CodePipeline pipeline which update the [App Stack](#3---app). These CI/CD resources only need to be deployed once per deployable branch, `main` in our case (we might choose to create adhoc environments by launching a new CI/CD stack targeting a different branch).
 
-These resources are deployed manually when changes occur. We could make yet another CodePipeline resource in the Setup section, but not today.
+These resources are deployed manually when changes occur. We could make yet another CodePipeline resource in the [Setup](#1---setup) section, but not today.
 
-* "deploy-cicd.sh" - Shell script to deploy this stack.
+* "deploy-cicd.sh" - Shell script to create/update this stack
 * "cicd.template.yml" - AWS resources for the CI/CD infrastructure
 
 ### 1 - Setup
 
-Finally, all of the above need some Roles to exist in the AWS accounts before we can run things with appropriate permisions. These roles are exported and used elsewhere. Elevated permissions is likely required to update these. This only needs to be created once, as roles and other resources can be shared by all CI/CD stacks for this application.
+Finally, all of the above need some Roles to exist in the AWS accounts before we can run things with appropriate permisions. These roles are exported and used elsewhere. Elevated permissions are required to update these (likely by an Infra Engineer). This only needs to be created once, as roles and other resources can be shared by all CI/CD stacks for this application.
 
-* "deploy-setup.sh"
+* "deploy-setup.sh" - Shell script to create/update this stack
 * "setup.template.yml" - AWS resources for the Setup infrastructure
 
-## Opportunities
+## Deploying an entirely new Javabuilder
+
+By setting the `TARGET_BRANCH` you can create a new CI/CD pipeline that watches for PR's and changes to the specified branch, deploying a Test and Production environment just like the standard pipeline.
+
+```
+TARGET_BRANCH=mybranch MODE=adhoc cicd/2-cicd/deploy-cicd.sh
+```
+
+## Deploying an Adhoc environment
+
+You can create an Adhoc environment by setting the `MODE` flag on the cicd deploy script. This will create a CI/CD pipeline that will watch for updates to your `TARGET_BRANCH`.
```

Comment:
Will the adhocs deploy to dev or prod? And what will the stack name be for a given adhoc?

Reply:
Following these instructions, they will deploy to prod. This would create a stack called 'javabuilder-mybranch' that would deploy another stack called 'javabuilder-adhoc-mybranch' [per this config](https://github.com/code-dot-org/javabuilder/blob/8664d0fc20851434f2edfa5ac037413ed4cb9507/cicd/2-cicd/cicd.template.yml#L313)

That said, I think the procedure to deploy to Dev would be:
- authenticate with Dev instead of prod
- change adhoc.config to use dev-code.org and the appropriate hosted zone ID
- ensure that the stack from "1-Setup" has been deployed to Dev.

If sticking adhocs in Dev would be the preference (makes sense to me) I can modify these instructions and make sure that works.

Reply:
I'm not sure if that's the preference or not, I'll leave it up to you whether you think that's valuable. Only potential reason to put them in dev is we do have account-wide limits on some things, but it's pretty hard to make a dev instance hit those limits

Reply:
I copied the template and moved it back to the root. I also renamed "deploy.sh" to "deploy-beta.sh"

The template within the cicd configuration includes the API>Api changes, so from here on these templates have diverged, but we can still maintain the beta environment.